<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ChirpConfigWidget</class>
 <widget class="QWidget" name="ChirpConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>843</width>
    <height>603</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="1,3">
   <item>
    <widget class="ChirpConfigPlot" name="chirpPlot" native="true"/>
   </item>
   <item>
    <widget class="QGroupBox" name="chirpConfigurationBox">
     <property name="toolTip">
      <string>Minimum protection and delay settings can be changed by editing the Blackchirp.conf file. See the chirpConfig section.</string>
     </property>
     <property name="title">
      <string>Chirp Configuration</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="addButton">
          <property name="text">
           <string> Add Segment</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/resources.qrc">
            <normaloff>:/icons/add.png</normaloff>:/icons/add.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="insertButton">
          <property name="text">
           <string> Insert Segment</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/resources.qrc">
            <normaloff>:/icons/go-last.png</normaloff>:/icons/go-last.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addEmptyButton">
          <property name="text">
           <string>Add Empty Segment</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/resources.qrc">
            <normaloff>:/icons/add.png</normaloff>:/icons/add.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="insertEmptyButton">
          <property name="text">
           <string>Insert Empty Segment</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/resources.qrc">
            <normaloff>:/icons/go-last.png</normaloff>:/icons/go-last.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="moveUpButton">
          <property name="text">
           <string> Move Up</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/resources.qrc">
            <normaloff>:/icons/arrow-up.png</normaloff>:/icons/arrow-up.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="moveDownButton">
          <property name="text">
           <string> Move Down</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/resources.qrc">
            <normaloff>:/icons/arrow-down.png</normaloff>:/icons/arrow-down.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="removeButton">
          <property name="text">
           <string> Remove</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/resources.qrc">
            <normaloff>:/icons/remove.png</normaloff>:/icons/remove.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="clearButton">
          <property name="text">
           <string> Clear All</string>
          </property>
          <property name="icon">
           <iconset resource="../../resources/resources.qrc">
            <normaloff>:/icons/clear.png</normaloff>:/icons/clear.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,1">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Current Chirp</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="currentChirpBox"/>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="applyToAllBox">
            <property name="text">
             <string>Apply to All</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableView" name="chirpTable">
          <property name="minimumSize">
           <size>
            <width>350</width>
            <height>0</height>
           </size>
          </property>
          <property name="gridStyle">
           <enum>Qt::NoPen</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="1" column="0">
         <widget class="QLabel" name="preChirpDelayLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Pre-Chirp Delay</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="protectionDelayLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Post-Chirp Protection</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="chirpsLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Chirps</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QSpinBox" name="chirpsSpinBox">
          <property name="toolTip">
           <string>Number of chirps in the AWG waveform.</string>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="chirpIntervalLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Chirp Interval</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QDoubleSpinBox" name="chirpIntervalDoubleSpinBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Time from beginning of one chirp to the beginning of the next.&lt;/p&gt;&lt;p&gt;Minimum is pre-chirp delay + chirp duration + protection delay + FID time + 4 Î¼s.&lt;/p&gt;&lt;p&gt;Setting the chirp interval below this value will cause problems, as the scope will miss chirps and become unsynschronized.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="suffix">
           <string/>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="minimum">
           <double>5.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>1.000000000000000</double>
          </property>
          <property name="value">
           <double>20.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="preChirpProtectionLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Pre-Chirp Protection</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="postChirpDelayLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Post-Chirp Delay</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="preChirpProtectionDoubleSpinBox">
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="preChirpDelayDoubleSpinBox">
          <property name="toolTip">
           <string>After t0 pulse (which sends TWT enable and protection signals), how long to wait before starting chirp.</string>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="postChirpDelayDoubleSpinBox">
          <property name="toolTip">
           <string>Following chirp, how long to wait before disabling TWT.</string>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="postChirpProtectionDoubleSpinBox">
          <property name="toolTip">
           <string>Following chirp, how long to wait before closing protection switch.</string>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ChirpConfigPlot</class>
   <extends>QWidget</extends>
   <header location="global">src/gui/plot/chirpconfigplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>addButton</tabstop>
  <tabstop>insertButton</tabstop>
  <tabstop>moveUpButton</tabstop>
  <tabstop>moveDownButton</tabstop>
  <tabstop>removeButton</tabstop>
  <tabstop>clearButton</tabstop>
  <tabstop>preChirpProtectionDoubleSpinBox</tabstop>
  <tabstop>preChirpDelayDoubleSpinBox</tabstop>
  <tabstop>postChirpProtectionDoubleSpinBox</tabstop>
  <tabstop>chirpsSpinBox</tabstop>
  <tabstop>chirpIntervalDoubleSpinBox</tabstop>
  <tabstop>chirpTable</tabstop>
 </tabstops>
 <resources>
  <include location="../../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
